.github/workflows/build-pack-release.yml

```
@@ -0,0 +1,96 @@
+name: Build-Pack-Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The version to tag the release with, e.g., 1.2.0, 1.3.0
+        required: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+    build-pack-release:
+        runs-on: ${{ matrix.os }}
+
+        env:
+            DOTNET_MULTILEVEL_LOOKUP: 1
+
+        strategy:
+            fail-fast: true
+            matrix:
+                os: [windows-latest]
+
+        steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+
+        - name: Configure AWS credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN_RELEASE }}
+            aws-region: us-west-2
+
+        - name: Get Strong-Naming Key
+          run: aws s3api get-object --bucket ${{ secrets.SNK_BUCKET }} --key awsxrayrecorder.dll.snk awsxrayrecorder.dll.snk
+          
+        - name: Install dependencies
+          run: dotnet restore .\sdk\AWSXRayRecorder.sln
+
+        - name: Build solution
+          run: dotnet build .\sdk\AWSXRayRecorder.sln /p:Configuration=Release --no-restore /p:AssemblyOriginatorKeyFile=${{ github.workspace }}\awsxrayrecorder.dll.snk
+
+        - name: Assume signer role
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.AWS_ARTIFACT_ACCESS_ROLE_ARN }}
+            aws-region: us-west-2
+
+        - name: Invoke Signing script
+          run: |
+            .\buildtools\sign_files.ps1 -Path .\sdk\src -Filter AWSXRayRecorder.*.dll -Recurse
+          env:
+            UNSIGNED_BUCKET: ${{ secrets.AWS_UNSIGNED_BUCKET_NAME }}
+            SIGNED_BUCKET: ${{ secrets.AWS_SIGNED_BUCKET_NAME }}
+
+        - name: Pack nugets
+          run: |
+            dotnet pack .\sdk\AWSXRayRecorder.sln  --no-build -c Release -o .\Deployment\nuget-packages
+            nuget pack .\sdk\AWSXRayRecorder.nuspec -OutputDirectory .\Deployment\nuget-packages -Exclude **
```

Would it be worth it to split this into 2 steps to let them each fail separately?

Otherwise, mostly a nit.

```suggestion
        - name: Dotnet pack nugets
          run: >
            dotnet pack .\sdk\AWSXRayRecorder.sln 
              --no-build
              -c Release
              -o .\Deployment\nuget-packages
        - name: Nuget pack nugets
          run: >
            nuget pack .\sdk\AWSXRayRecorder.nuspec
              -OutputDirectory .\Deployment\nuget-packages
              -Exclude **
```

---

Sounds good. In that case is it still possible to separate arguments on a new line? According to [this StackOverflow post](https://stackoverflow.com/questions/605686/how-to-write-a-multiline-command) you can do it like this but I am not confident in it 😅

```suggestion
        - name: Pack nugets
          run: |
            dotnet pack .\sdk\AWSXRayRecorder.sln  ^
              --no-build ^
              -c Release ^
              -o .\Deployment\nuget-packages
            nuget pack .\sdk\AWSXRayRecorder.nuspec ^
              -OutputDirectory .\Deployment\nuget-packages ^
              -Exclude **
```

---

I'm not confident either. 😅 
As per this [yaml multiline doc](https://yaml-multiline.info/), "To get a newline using the folded style, leave a blank line by putting two newlines in.".
Then I think these command would look like this:
```yaml
- name: Pack nugets
   run: >
      dotnet pack
        .\sdk\AWSXRayRecorder.sln
        --no-build
        -c Release
        -o .\Deployment\nuget-packages

      nuget pack
        .\sdk\AWSXRayRecorder.nuspec
        -OutputDirectory .\Deployment\nuget-packages
        -Exclude **
```